### PR TITLE
Migrate to Helm Chart

### PR DESCRIPTION
**Pull request by kezhenxu94**
Migrate to Helm Chart for several reasons:

- Better leverage our ecosystem projects like `skywalking-kubernetes` and `skywalking-swck`, which are recommended for our users and they are more realistic in real use case, and thus reduce duplicate works to replicate the same features in pure Kubernetes resources.
- More flexible because we can have fine-grained conditions in the resources with Helm Chart templates.
- Don't rely on users' local installation (version) of tools such as `istioctl`, the only requirement is Helm.
- Easy to add features that are supported in our official eco-system projects, such as, adding new storage options (such as `postgresql`), will be much more easier.

### CODE REVIEW

**Comment by kezhenxu94** [comment]
I'm running the postgresql demos in my environment and will merge this when I think it's ready

**Comment by kezhenxu94** [comment]
@innerpeacez welcome to try this and advice

**Comment by wu-sheng** [comment]
I love this idea. Using helm/operator is much better than writing all these again.
BTW, we should consider a new helm release as many changes have been made.

**Comment by kezhenxu94** [comment]
> BTW, we should consider a new helm release as many changes have been made.

You mean `skywalking-kubernetes` repo right?

**Comment by wu-sheng** [comment]
> You mean skywalking-kubernetes repo right?

Yes, that. 
___

How do we lock the version of helm repo? I can't find the relative changes of make.in in the root.

**Comment by innerpeacez** [comment]
> @innerpeacez welcome to try this and advice

WOW!!! looking forward to this.

File: deploy/platform/kubernetes/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+apiVersion: v2
+name: skywalking-showcase
+home: https://skywalking.apache.org
+version: 0.0.0
+description: Apache SkyWalking showcase
+icon: https://raw.githubusercontent.com/apache/skywalking-kubernetes/master/logo/sw-logo-for-chart.jpg
+sources:
+  - https://github.com/apache/skywalking-showcase
+dependencies:
```
Comment (kezhenxu94) [inline deploy/platform/kubernetes/Chart.yaml:24]: > How do we lock the version of helm repo? I can't find the relative changes of make.in in the root.


@wu-sheng most of the versions are still maintained in the `Makefile.in` file, only the Helm Chart versions in this files are different, they have to be changed in this file, no variable substitutions can be done in `Chart.yaml` file here

Comment (kezhenxu94) [reply deploy/platform/kubernetes/Chart.yaml:24]: From the content here, these versions are Helm Chart versions themselves, not the app version (like `OAP` version), so in fact the previous versions (before this PR) are still the same as before

Comment (wu-sheng) [reply deploy/platform/kubernetes/Chart.yaml:24]: OK, as this is a new place. Let's add a doc about how to bump up component versions, especially for SkyWalking components. 

@wankai123 @kezhenxu94 Another thing about OTEL collector version, we are not bumping up for a while, there is a big gap, we should doc them and try to bump them.

Comment (innerpeacez) [reply deploy/platform/kubernetes/Chart.yaml:24]: > @wankai123 @kezhenxu94 Another thing about OTEL collector version, we are not bumping up for a while, there is a big gap, we should doc them and try to bump them.

opentelemetry-collector:0.72.0
kube-state-metrics:v2.8.0

I'm using this version with no problems

File: deploy/platform/kubernetes/templates/feature-event/permissions.yaml [hunk 2]
```diff
@@ -32,5 +34,6 @@ roleRef:
   name: view
 subjects:
   - kind: ServiceAccount
-    namespace: ${NAMESPACE}
+    namespace: {{ .Release.Namespace }}
```
Comment (innerpeacez) [inline deploy/platform/kubernetes/templates/feature-event/permissions.yaml:37]: Does the namespace need to be explicitly specified? This should be the default.

File: deploy/platform/kubernetes/Makefile.in [hunk 3]
```diff
@@ -18,8 +18,8 @@
 
 .EXPORT_ALL_VARIABLES:
 
-NAMESPACE ?= default
+NAMESPACE ?= skywalking-showcase
```
Comment (kezhenxu94) [inline deploy/platform/kubernetes/Makefile.in:21]: > Does the namespace need to be explicitly specified? This should be the default.

@innerpeacez we override the namespace here to `skywalking-showcase`

Comment (innerpeacez) [reply deploy/platform/kubernetes/Makefile.in:21]: helm install -n `namespace`, which seems to be the default behavior of `-n`

Comment (innerpeacez) [reply deploy/platform/kubernetes/Makefile.in:21]: I think it is only necessary to define the namespace in the Makefile. There is no need to explicitly specify the namespace in k8s resouces yaml. The namespace will use .Release.Namespace by default

Comment (kezhenxu94) [reply deploy/platform/kubernetes/Makefile.in:21]: OK I got your point now, make sense to me.

File: deploy/platform/kubernetes/Chart.yaml (hunk 1 above)
Comment (wu-sheng) [reply deploy/platform/kubernetes/Chart.yaml:24]: @kezhenxu94 Could we upgrade this as well?

Comment (kezhenxu94) [reply deploy/platform/kubernetes/Chart.yaml:24]: > @kezhenxu94 Could we upgrade this as well?

Sure

Comment (kezhenxu94) [reply deploy/platform/kubernetes/Chart.yaml:24]: Currently the kube-state-metrics is latest 5.0.1

Comment (kezhenxu94) [reply deploy/platform/kubernetes/Chart.yaml:24]: I can update the otel version

File: deploy/platform/kubernetes/templates/feature-event/permissions.yaml (hunk 2 above)
Comment (kezhenxu94) [reply deploy/platform/kubernetes/templates/feature-event/permissions.yaml:37]: > Does the namespace need to be explicitly specified? This should be the default.

I think you take it wrong, this `namespace` is in `subjects`, not in `metadata`, in `metadata` the default value is `{{ .Release.Namespace }}`, but in `subjects`, there is no default value for `namespace` and the `namespace` is required for `subjects` so we have to set it explicitly here

**Pull request merged**